### PR TITLE
Issue #71 Manage organizations backend

### PR DESCRIPTION
**Pull request by RobertBeaudenon**
User Story 1.6
Provided the following functionalities:
- Created a model responsible of holding organization information
- Created an API endpoint that allows user to perform CRUD operations on organizations
- Added custom permissions on the API endpoint
  - A System Admin is allowed  to: Create, Update, Delete, Retrieve and Organization
  - An  Inventory Manager is allowed to: Update and Retrieve and Organization
- Provided test cases for that

### CODE REVIEW

File: server/django_server/permissions.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        This can be used either inside the permission_class directly or you can call it from other permission files
+        :param request:
+        :param view:
+        :return: True/False : Whether the user is a Inventory Manager or Not
+        """
+        user = CustomUser.objects.get(user_name=request.user)
+        if user.role == 'IM':
+            return True
+        else:
+            return False
```
Comment (makchamp) [inline server/django_server/permissions.py:38]: You can return the expression itself to save loc
return user.role == 'IM'

File: server/organization/permissions.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+             :param request:
+             :param view:
+             :return: True/False : Whether the user is allowed to perform CRUD
+             """
+        if view.action in ['list', 'retrieve', 'update', 'partial_update']:
+            return isSystemAdmin.has_permission(None, request, None) or isInventoryManager.has_permission(None, request, None)
+        elif view.action in ['create', 'destroy']:
+            return isSystemAdmin.has_permission(None, request, None)
+        else:
+            return False
```
Comment (makchamp) [inline server/organization/permissions.py:20]: same here you can return the expression itself

File: server/organization/tests.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from django.test import TestCase
+from django.test.client import Client as HttpClient
+from rest_framework import status
+from rest_framework.test import APITestCase
+from rest_framework.authtoken.models import Token
+from rest_framework.test import APIClient
+from user_account.models import CustomUser
+
+
+class OpenRegistrationTestCase(APITestCase):
```
Comment (makchamp) [inline server/organization/tests.py:11]: This test case is dummy for an endpoint that will not be used in production. Maybe add these tests to the actual Registration endpoint test case class or create a new one just for organization ?

**Review by makchamp — CHANGES_REQUESTED** [review]
Good change. Will merge once the errors are cleaned up.
However I would like to raise one point. I did not see how you are storing which user belongs to which organization. An option would be to give the user model a reference to the organization model (through a foreign key or something). Maybe we should reference our design docs/diagrams for this.

File: server/django_server/permissions.py (hunk 1 above)
Comment (RobertBeaudenon) [reply server/django_server/permissions.py:38]: Yes, I'll fix it.

File: server/organization/permissions.py (hunk 2 above)
Comment (RobertBeaudenon) [reply server/organization/permissions.py:20]: In this case, i cant since im using elif.

File: server/organization/tests.py (hunk 3 above)
Comment (RobertBeaudenon) [reply server/organization/tests.py:11]: it's for organization I just copied the structure but forgot to change the name of the class hahaha

**Review by makchamp — APPROVED** [review]
Thanks @RobertBeaudenon for implementing the requested changes.
As for linking the user and organization models, it was discussed that further work on the matter will be done in a future PR.

**Pull request merged**